internal/mcp-router/request_handlers.go

```diff
@@ -257,6 +257,18 @@ func (s *ExtProcServer) HandleToolCall(ctx context.Context, mcpReq *MCPRequest)
 	headers.WithMCPSession(remoteMCPSeverSession)
 	// reset the host name now we have identifed the correct tool and backend
 	headers.WithAuthority(serverInfo.Hostname)
+
+	// add authorization headers for backends that require auth
+	mcpServerConfig := s.RoutingConfig.GetServerConfigByName(serverInfo.Name)
```

Comment:
I don't think we want to do this. The credential provided in the MCPServer resource is intended for use by the broker to initialise , list tools and listen for global notifications. With this change we allow an external client to use that credential. I think it would be better to use an authpolicy for the route that loads the needed credential during the tools call. This change will overwrite any Authorization set by the client by an AuthPolicy if the backend mcpserver has specified a credential.

See example here of fetching a key from vault https://github.com/kagenti/mcp-gateway/blob/main/config/samples/oauth-token-exchange/tools-call-auth.yaml#L18